pkg/model/components/cilium.go

```diff
@@ -168,5 +168,31 @@ func (b *CiliumOptionsBuilder) BuildOptions(o interface{}) error {
 		}
 	}
 
+	ingress := c.Ingress
+	if ingress != nil {
+		if ingress.Enabled == nil {
+			ingress.Enabled = fi.PtrTo(true)
+		}
+		if ingress.EnforceHttps == nil {
+			ingress.EnforceHttps = fi.PtrTo(true)
+		}
+		if ingress.EnableSecretsSync == nil {
+			ingress.EnableSecretsSync = fi.PtrTo(true)
+		}
+		if ingress.LoadBalancerAnnotationPrefixes == "" {
+			ingress.LoadBalancerAnnotationPrefixes = "service.beta.kubernetes.io service.kubernetes.io cloud.google.com"
+		}
+		if ingress.DefaultLoadBalancerMode == "" {
+			ingress.DefaultLoadBalancerMode = "dedicated"
+		}
+		if ingress.SharedLoadBalancerServiceName == "" {
+			ingress.SharedLoadBalancerServiceName = "cilium-ingress"
+		}
+	} else {
+		c.Ingress = &kops.CiliumIngressSpec{
+			Enabled: fi.PtrTo(false),
+		}
+	}
```

Comment:
Can we try something like this, to not populate all the parameters if ingress is not enabled?
```suggestion
	if ingress == nil {
		c.Ingress = &kops.CiliumIngressSpec{
			Enabled: fi.PtrTo(false),
		}
	}
	if ingress.Enabled == nil {
		ingress.Enabled = fi.PtrTo(true)
	}
	if fi.ValueOf(ingress.Enabled ) {
		if ingress.EnforceHttps == nil {
			ingress.EnforceHttps = fi.PtrTo(true)
		}
		if ingress.EnableSecretsSync == nil {
			ingress.EnableSecretsSync = fi.PtrTo(true)
		}
		if ingress.LoadBalancerAnnotationPrefixes == "" {
			ingress.LoadBalancerAnnotationPrefixes = "service.beta.kubernetes.io service.kubernetes.io cloud.google.com"
		}
		if ingress.DefaultLoadBalancerMode == "" {
			ingress.DefaultLoadBalancerMode = "dedicated"
		}
		if ingress.SharedLoadBalancerServiceName == "" {
			ingress.SharedLoadBalancerServiceName = "cilium-ingress"
		}
	}
```

Reply:
Looks good :) I'll clean up the commits in the end

Reply:
@hakman looks like this results in nil pointer dereference :(

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kops/15795/pull-kops-verify-golangci-lint/1694393521546465280

Any idea how to fix this?

Reply:
Yes, see my suggestion. You will also need to run hack/update-expected.